### PR TITLE
added cnet fixture

### PR DESCRIPTION
**Pull request by Kelvinrr**
## Description

This adds a basic three image cnet fixture. This should open the door for basic cnet related tests. As most app tests are very simple and the heavy lifting is often covered by unittests, I imagine one fixture would be enough to cover a lot of cases.  

Footprints: 

Screen Shot 2020-02-11 at 1 25 02 PM

Control points were `autoseed`-ed in. 

## Motivation and Context

get cnet based app tests going. 

## How Has This Been Tested?

Hand verified the network through qnet. No real tests for this fixture yet. I'm okay with leaving this open until we use it in an actual app test if it's an issue. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/src/base/objs/Cube/Cube.cpp [hunk 1]
```diff
@@ -107,7 +107,6 @@ namespace Isis {
    * @param fileName Name of the cube file to open. Environment
    *     variables in the filename will be automatically expanded.
    * @param label PVL label to use when initializing cube 
-   * @param isd Ale compatible ISD to be used for initing spice data
```
Comment (krlberry) [inline isis/src/base/objs/Cube/Cube.cpp:110]: Why was this line removed? It looks like it's still used in the function.

File: isis/src/base/objs/Cube/Cube.cpp [hunk 2]
```diff
@@ -118,7 +117,20 @@ namespace Isis {
     close();
     open(fileName.toString(), access);
   }
-
+
```
Comment (krlberry) [inline isis/src/base/objs/Cube/Cube.cpp:121]: Add doxygen documentation, please.

File: isis/src/base/objs/Cube/Cube.cpp [hunk 3]
```diff
@@ -118,7 +117,20 @@ namespace Isis {
...
+    std::ifstream labelStream(labelFile.expanded().toStdString());
+    
+    Pvl label;
+    nlohmann::json isd;
+    
+    isdStream >> isd;
+    labelStream >> label;
+    
+    fromIsd(fileName, label, isd, access);
+    reopen("rw");
```
Comment (krlberry) [inline isis/src/base/objs/Cube/Cube.cpp:143]: Why is it necessary to `reopen("rw")` right after calling `fromIsd`, which closes and reopens the cube in its last 2 lines?

File: isis/tests/Fixtures.cpp [hunk 4]
```diff
@@ -2,203 +2,68 @@
...
-        "Velocities": [
-          [10,10,10]
-        ]
-      }
-    })");
-
-    // call constructor
-    Cube testCube;
+  void DefaultCube::SetUp() {
+    std::ifstream isdFile("data/defaultImage/defaultCube.isd");
```
Comment (krlberry) [inline isis/tests/Fixtures.cpp:6]: :+1: On moving these into files. This improves the readability of the tests!

File: isis/tests/Fixtures.cpp [hunk 5]
```diff
@@ -2,203 +2,68 @@
...
     
-    testCube.fromIsd(tempFile.fileName() + ".cub", label, testIsd, "rw");
+    isdFile >> isd;
+    cubeLabel >> label;
+
+    testCube->fromIsd(tempFile.fileName() + ".cub", label, isd, "rw");
 
     // reopen to write out new data 
-    testCube.reopen("rw");
+    testCube->reopen("rw");
```
Comment (krlberry) [inline isis/tests/Fixtures.cpp]: Now that I'm looking at this again, I'm confused by why, first `fromIsd` closes and reopens the cube and then the cube needs to be reopened again here? Does leaving this out break it?

**Review by krlberry — CHANGES_REQUESTED** [review]
Looks good! Had a couple of questions and a couple of minor documentation things. (The 'Request changes' only refers to the documentation request.)

File: isis/src/base/objs/Cube/Cube.cpp (hunk 3 above)
Comment (Kelvinrr) [reply isis/src/base/objs/Cube/Cube.cpp:143]: Good catch :+1: I imagine it was moved into fromIsd and wasn't removed here.

File: isis/tests/Fixtures.cpp (hunk 5 above)
Comment (Kelvinrr) [reply isis/tests/Fixtures.cpp]: It should be fine, seems like a missed detail when converting the fixture logic over to fromIsd. reopen was added to fromIsd but wasn't removed from the fixtures.

File: isis/src/base/objs/Cube/Cube.cpp (hunk 1 above)
Comment (Kelvinrr) [reply isis/src/base/objs/Cube/Cube.cpp:110]: No idea, do not recall removing it. Adding it back in.

**Review by krlberry — DISMISSED** [review]
:+1:

File: isis/src/base/objs/Cube/Cube.cpp [hunk 6]
```diff
@@ -106,7 +106,7 @@ namespace Isis {
    *
    * @param fileName Name of the cube file to open. Environment
    *     variables in the filename will be automatically expanded.
-   * @param label PVL label object to use  
+   * @param label PVL label object reprenting the new Cube label
```
Comment (krlberry) [inline isis/src/base/objs/Cube/Cube.cpp]: reprenting -> representing

**Review by krlberry — DISMISSED** [review]

**Review by krlberry — APPROVED** [review]
